### DISCUSSION TITLE
Being detected with send keys but not with manual input

### DISCUSSION THREAD

**Discussion by versionDefect** [ultrafunkamsterdam/undetected-chromedriver#1074] [General]
So my work makes me do this tedious process of checking my schedule so I automated it. Worked for months but for some reason, my code isn't functioning anymore. 

It detected when it automatically inputted the data, but NOT when I manually inputted it thru the same UC selenium browser. What could I do to make it undetectable? 

I have a bajillion flags for headless mode (before GChrome updated headless mode to be nearly undetectable) but with or without them still does not function properly. 

EDIT: forgot to mention. I even tested something with a command like 

```actions.pause(4).move_to_element.pause(1).click().perform()``` to select the username and password boxes and even slowly streaming in my username and password but still getting detected. 

Thoughts?

EDIT:! I FIXED IT. But I do think I should maybe make an issue. For some reason, Downgrading from the newest version to 3.2.1 fixed it. Idk why or what changed but my employer's website likes me again. what an odd issue.

For some reason, versions 3.4.0 -> 3.4.2 break my code and don't run at all. But then in 3.4.4 (3.4.3 doesn't exist lol) I get detected.

But downgrading to version 3.2.0 / 3.2.1 works perfectly!

downgrading didn't work perfectly but it is fixable! If you're not using headless then it SHOULD work on the newest version but if you are using headless your code will NOT work. In addition to downgrading UC you should also downgrade chrome from 110 -> 109

**Comment by pstefa1707** [comment]
Subscribed to this as I'm facing the same problem.

One work around I found is to use the `--auto-open-devtools-for-tabs` to open dev tools. 
For some weird reason this seemed to work for a couple hours? Then the page started detecting the chromedriver again.

Anyways, keen to see what comes of this. I also noticed the page console-logging `Undetected Chromedriver 1337`, which I find quite funny, seems like someone's detecting it!

**Reply by versionDefect** [reply]
OKAY I FIXED IT. But I do think I should maybe make an issue. For some reason, Downgrading from the newest version to 3.2.1 fixed it. Idk why or what changed but my employer website likes me again. what an odd issue. 

For some reason, versions 3.4.0 -> 3.4.2 break my code and doesn't run at all. But then in 3.4.4 (3.4.3 doesn't exist lol) I get detected. 

But downgrading to version 3.2.0 / 3.2.1 works perfectly!

EDIT: downgrading didn't work perfectly but it is fixable! If you're not using headless then it SHOULD work but if you are using headless your code will NOT work. 

In addition to downgrading UC you should also downgrade chrome from 110 -> 109

**Reply by AlexPaiva** [reply]
Tested 3.2.1 with chrome 110 and also worked perfectly. Have you tried this?

**Reply by versionDefect** [reply]
> Tested 3.2.1 with chrome 110 and also worked perfectly. Have you tried this?

It worked but not in headless mode. could be my settings for headless honestly. But i didn't wanna tinker my recipe. i had FINALLY got it to work

**Reply by hamik112** [reply]
Headless mode is most likely still not 100% similiar to non-headless mode. It's pretyy new so it's likely its not exactly stable

**Reply by versionDefect** [reply]
> Headless mode is most likely still not 100% similiar to non-headless mode. It's pretyy new so it's likely its not exactly stable

headless is a bit of an odd thing honestly. I threw a slurry of random flags until headless was undetectable. My code has been running for ages now. Considering reoptimizing it. don't have the time tho lol.

**Reply by AlexPaiva** [reply]
> > Tested 3.2.1 with chrome 110 and also worked perfectly. Have you tried this?
> 
> It worked but not in headless mode. could be my settings for headless honestly. But i didn't wanna tinker my recipe. i had FINALLY got it to work

What recipe exactly if you don't mind me asking?

**Reply by pstefa1707** [reply]
> OKAY I FIXED IT. But I do think I should maybe make an issue. For some reason, Downgrading from the newest version to 3.2.1 fixed it. Idk why or what changed but my employer website likes me again. what an odd issue.

Thank you for the information! I'll give this a try today :)

> headless is a bit of an odd thing honestly. I threw a slurry of random flags until headless was undetectable. My code has been running for ages now. Considering reoptimizing it. don't have the time tho lol.

May you share your flags if you don't mind me asking? 🙏

**Reply by versionDefect** [reply]
> > OKAY I FIXED IT. But I do think I should maybe make an issue. For some reason, Downgrading from the newest version to 3.2.1 fixed it. Idk why or what changed but my employer website likes me again. what an odd issue.
> 
> 
> 
> Thank you for the information! I'll give this a try today :)
> 
> 
> 
> > headless is a bit of an odd thing honestly. I threw a slurry of random flags until headless was undetectable. My code has been running for ages now. Considering reoptimizing it. don't have the time tho lol.
> 
> 
> 
> May you share your flags if you don't mind me asking? 🙏

'''
options = uc.ChromeOptions()
options.add_argument("--password-store=basic")
if config['options']['headless'] == 'True':
	options.headless = True
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--incognito")
options.add_argument("--disable-extensions")
options.add_argument("--enable-automation")
options.add_argument("--disable-browser-side-navigation")
options.add_argument("--disable-web-security")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--disable-infobars")
options.add_argument("--disable-gpu")
options.add_argument("--disable-setuid-sandbox")
options.add_argument("--disable-software-rasterizer")

options.add_experimental_option(
    "prefs",
    {
        "credentials_enable_service": False,
        "profile.password_manager_enabled": False,
    },
)

print('- Finished importing packages')

'''
this was for python. I also don't know how much of this is necessary but it's good enough to get through to my fortune 500 company LOL.

**Reply by pstefa1707** [reply]
My chrome v110 crashes with UC 3.2.1 :( [Note i'm on mac m1, arm64 install]

Also, didn't fix my issues while it wasn't crashing, I wonder how they're noticing it, I've accepted defeat I think hahaha

**Reply by versionDefect** [reply]
> My chrome v110 crashes with UC 3.2.1 :( [Note i'm on mac m1, arm64 install]
> 
> 
> 
> Also, didn't fix my issues while it wasn't crashing, I wonder how they're noticing it, I've accepted defeat I think hahaha 

Hmm. This convo is more so for people getting detected for sendKeys.

if you've downgraded make sure you specify what chrome version you use 

driver = uc.Chrome( version_main = 95 ) replace 95 with what ever you're using as your version. Although i don't really know how you're going to install an older version of chrome without it updating automatically. i'm running on ubuntu so i just disabled it there.

**Reply by AlexPaiva** [reply]
> > > OKAY I FIXED IT. But I do think I should maybe make an issue. For some reason, Downgrading from the newest version to 3.2.1 fixed it. Idk why or what changed but my employer website likes me again. what an odd issue.
> > 
> > 
> > Thank you for the information! I'll give this a try today :)
> > > headless is a bit of an odd thing honestly. I threw a slurry of random flags until headless was undetectable. My code has been running for ages now. Considering reoptimizing it. don't have the time tho lol.
> > 
> > 
> > May you share your flags if you don't mind me asking? 🙏
> 
> ''' options = uc.ChromeOptions() options.add_argument("--password-store=basic") if config['options']['headless'] == 'True': options.headless = True options.add_argument("--no-sandbox") options.add_argument("--disable-dev-shm-usage") options.add_argument("--incognito") options.add_argument("--disable-extensions") options.add_argument("--enable-automation") options.add_argument("--disable-browser-side-navigation") options.add_argument("--disable-web-security") options.add_argument("--disable-dev-shm-usage") options.add_argument("--disable-infobars") options.add_argument("--disable-gpu") options.add_argument("--disable-setuid-sandbox") options.add_argument("--disable-software-rasterizer")
> 
> options.add_experimental_option( "prefs", { "credentials_enable_service": False, "profile.password_manager_enabled": False, }, )
> 
> print('- Finished importing packages')
> 
> ''' this was for python. I also don't know how much of this is necessary but it's good enough to get through to my fortune 500 company LOL.

options.add_experimental_option(
"prefs",
{
"credentials_enable_service": False,
"profile.password_manager_enabled": False,
},

this is for what exactly?

**Reply by versionDefect** [reply]
This was just something I had for my own reasons lol. My company wants me to signup for FIDO (using like Windows Hello or macOS authentication etc). Incognito mode makes my company not ask that question. May cause issues depending on your site. You don't really need that.

**Comment by zikmirpassiv** [comment]
Super helpful! Right after the update it broke the send_keys(). Thanks again for the input, this should be an issue.

**Reply by versionDefect** [reply]
Made an issue about it but seems like owner marked it as closed. This is still up so if needed people can view this lol.

**Comment by AlexPaiva** [comment]
Just got detected by google using the previous version, anyone else?

**Comment by maximusrus** [comment]
Got detected too even without headless mode. Tried settings from the post above and downgrade undetected-chrome version but it didn’t help. Hope anyone can help or suggest something. 

Did not try downgrade chrome version though

**Reply by maximusrus** [reply]
I'm also detected by click, click_safe() and send_keys(Keys.ENTER) methods =(. Manual pressing the button does not make me a bot

**Reply by versionDefect** [reply]
> I'm also detected by click, click_safe() and send_keys(Keys.ENTER) methods =(. Manual pressing the button does not make me a bot

i made an issue about this ages ago but the dev closed it 🤷‍♀️

**Reply by maximusrus** [reply]
How do you identify that you were caught as a bot?
Maybe it will help me to try more things to avoid detection.

**Reply by versionDefect** [reply]
> How do you identify that you were caught as a bot?
> 
> Maybe it will help me to try more things to avoid detection. 

In my case my website just doesn't let me log in. it says "Something went wrong. Try again later"

**Reply by maximusrus** [reply]
In my specific case most of the times I cannot login (sometimes I can on hundreds of attempts) but the weird scenario I found is that when I press forgot password I do not get the link on my email. Manually I easily receive this.

**Comment by ypsik** [comment]
same issue, detected after click, click_safe() and send_keys(), on akamai protected site, after registration getting 403 response

**Reply by versionDefect** [reply]
> same issue, detected after click, click_safe() and send_keys(), on akamai protected site, after registration getting 403 response

Made an issue with the dev. They closed it workout considering it.

**Reply by rockccf** [reply]
Has anyone found a way to fix this?

**Reply by versionDefect** [reply]
The project I was working on is dead but, the main solution I found was using an older version of UDCD
